compiler/resolution/functionResolution.cpp

```diff
@@ -4663,6 +4670,37 @@ resolveNew(CallExpr* call)
     call->baseExpr = arg;
     parent_insert_help(call, call->baseExpr);
 
+    if (initCall) {
+      // 4: insert the allocation for the instance and pass that in as an
+      // argument if we're making a call to an initializer
+      Type* typeToNew = toReplace->var->typeInfo();
+      VarSymbol* new_temp = newTemp(typeToNew);
+      if (typeToNew->symbol->hasFlag(FLAG_GENERIC)) {
+        USR_FATAL(call, "Sorry, new style initializers don't work with generics yet.  Stay tuned!");
+      }
+      DefExpr* def = new DefExpr(new_temp);
+      Expr* insertBeforeMe = ((!isBlockStmt(call->parentExpr)) ?
+                              call->parentExpr : call);
+      insertBeforeMe->insertBefore(def);
+      resolveExpr(def);
+
+      if (!isRecord(typeToNew) && !isUnion(typeToNew)) {
+        BlockStmt* allocCallBlock = new BlockStmt();
```

Comment:
The callChplHereAlloc return result is not normalized.  Basically, inserting a blockStmt allowed me to normalize and resolve all these new statements in one go instead of having to call them all individually.